src/partitionAccessory.ts

```diff
@@ -128,19 +129,31 @@ export class EnvisalinkPartitionAccessory {
                 break;
             case 'disarmed':
             case 'notready':
-            case 'failedarm':
+            case 'failedtoarm':
             case 'useropening':
+            case 'specialopening':
                 currentState = this.platform.Characteristic.SecuritySystemCurrentState.DISARMED;
                 obstructionDetected = true;
                 break;
             case 'exitdelay':
             case 'entrydelay':
             case 'arming':
             case 'busy':
+            case 'troubleledoff':
                 break;
-            default:
+            case 'troubleledon':
+                this.platform.log.info(`Ignoring status '${this.partition.status.text}' (${this.partition.status.description})`);
+                break;
+            case 'ready':
+            case 'readyforce':
                 currentState = this.platform.Characteristic.SecuritySystemCurrentState.DISARMED;
                 targetState = this.platform.Characteristic.SecuritySystemTargetState.DISARM;
+                break;
+            default:
+                // If we do not recognise the security panel state then we should not assume
+                // that status is DISARMED.  Leave it unchanged and log warning message.  With
+                // luck user will report this and provide debug trace.
+                this.platform.log.warn(`Ignoring status '${this.partition.status.text}' (${this.partition.status.description})${REPORT_ERROR_TXT}`);
```

Comment:
I think this could be problematic for a default, but I'll publish as a beta and we can do some testing.